packages/plugin-core/src/commands/MergeNoteCommand.ts

```diff
@@ -0,0 +1,453 @@
+   * @param destNote Dest note
+   */
+  private async updateBacklinks(opts: {
+    sourceNote: NoteProps;
+    destNote: NoteProps;
+  }) {
+    const ctx = "MergeNoteCommand:updateBacklinks";
+    const { sourceNote, destNote } = opts;
+
+    // grab all backlinks from source note
+    const sourceBacklinks = sourceNote.links.filter((link) => {
+      return link.type === "backlink";
+    });
+
+    // scrub through the backlinks and all notes that need to be updated
+    const noteIDsToUpdate = Array.from(
+      new Set(
+        sourceBacklinks
+          .map((backlink) => backlink.from.id)
+          .filter((ent): ent is string => ent !== undefined)
+      )
+    );
+
+    // for each note that needs to be updated,
+    // find all links that need to be updated from end to front.
+    // then update them.
+    let noteChangeEntries: NoteChangeEntry[] = [];
+    await asyncLoopOneAtATime(noteIDsToUpdate, async (id) => {
+      try {
+        const changed = await this.updateLinkInNote({
+          sourceNote,
+          destNote,
+          id,
+        });
+        noteChangeEntries = noteChangeEntries.concat(changed);
+      } catch (error) {
+        this.L.error({ ctx, error });
+      }
+    });
+    return noteChangeEntries;
+  }
+
+  /**
+   * Given a source note, delete it
+   * @param sourceNote source note
+   */
+  private async deleteSource(opts: { sourceNote: NoteProps }) {
+    const ctx = `${this.key}:deleteSource`;
+    const { sourceNote } = opts;
+    try {
+      const deleteResp = await this.extension
+        .getEngine()
+        .deleteNote(sourceNote.id);
+      if (deleteResp.data) {
+        return deleteResp.data;
+      } else {
+        // This is very unlikely given a sound engine state.
+        // log it and gracefully return
+        return [];
+      }
+    } catch (error) {
+      this.L.error({ ctx, error });
+      return [];
+    }
+  }
+
+  async execute(opts: CommandOpts): Promise<CommandOutput> {
+    const ctx = "MergeNoteCommand";
+    this.L.info({ ctx, msg: "execute" });
+    const { sourceNote, destNote } = opts;
+
+    if (destNote === undefined) {
+      vscode.window.showWarningMessage("Merge destination not selected");
+      return {
+        ...opts,
+        changed: [],
+      };
+    }
+
+    // opts.notes should always have at most one element since we don't allow multiple destinations.
+    if (sourceNote === undefined) {
+      return {
+        ...opts,
+        changed: [],
+      };
+    }
+
+    const appendNoteChanges = await this.appendNote({ sourceNote, destNote });
+    const updateBacklinksChanges = await this.updateBacklinks({
+      sourceNote,
+      destNote,
+    });
+    const deleteSourceChanges = await this.deleteSource({ sourceNote });
+
+    const noteChangeEntries = [
+      ...appendNoteChanges,
+      ...updateBacklinksChanges,
+      ...deleteSourceChanges,
+    ];
+
+    // open the destination note
+    await this.extension.wsUtils.openNote(destNote);
```

Comment:
do we need to close the source note?

Reply:
I initially thought it'd be nice if the deleted note still stuck around in the editor, but it'd be better to close it to be consistent with the behavior of `Dendron: Delete` now that I think about it. Also better to clean up a note in the editor that isn't in the store to prevent the engine going out of sync potentially.